### ISSUE TITLE
Status line functions not autoloading?

### ISSUE THREAD

**Issue by sethen** [nickspoons/vim-sharpenup#3]
Trying to use the status line features but for some reason I am getting an error in Neovim that looks like:

```
E117: Unknown function: sharpenup#statusline#Build
E15: Invalid expression: sharpenup#statusline#Build()
```

Is it possible that this function is not being autoloaded correctly?

**Comment by nickspoons** [comment]
Oh I'm sorry, for some reason I didn't receive any notification about this issue!

How are you using the function?

Is it possible that your statusline is being built before this plugin has been added to your runtimepath?

**Comment by nickspoons** [comment]
Trying a direct mention, in case you're also not getting notifications @sethen

**Comment by nickspoons** [comment]
> Is it possible that your statusline is being built before this plugin has been added to your runtimepath?

Ah yeah I think that'll be it. I glanced at your dotfiles, and you put almost everything between `call plug#begin(...)` and `call plug#end()`. vim-plug doesn't set the runtimepath for plugins until `plug#end()` is called.

If you move your statusline configuration down to _after_ `call plug#end()`, your error should be resolved.

**Comment by sethen** [comment]
Hey sorry for the late reply.

Yes, that did it for me... However I am trying to show this status in a different way.  I have the following code in `init.vim`:

```
	function! LightlineSharpenUp()
		if IsIgnoringStatus()
			return ''
		endif

		return sharpenup#statusline#Build()
	endfunction
```
This essentially looks at the files I want to be ignored and does not load the plugin.  I only want this plugin to load when it's necessary (only C# projects).

This works for other things I have in my lightline.  For some reason for Sharpen Up it just outputs a bunch of gibberish like so:

![Screenshot from 2020-04-08 16-24-25](https://user-images.githubusercontent.com/1493530/78830102-6d6b2000-79b5-11ea-970b-6238d274d210.png)

Any thoughts?

**Comment by nickspoons** [comment]
The `sharpenup#statusline#Build()` line is generating statusline "expressions", which are evaluated at runtime. So that big line of output contains 3 sections, each of which can contain content and highlight groups. Only one of them will ever contain content at a time though - this is how dynamically highlighted statuslines are built up (have a look at what lightline is generating: `:echo &statusline`).

Since you are more actively integrating your lightline functions, you don't need any of that, you just need the status itself. So in your function, replace `return sharpenup#statusline#Build()` with:

```vim
return sharpenup#statusline#GetStatus().Text
```

**Comment by sethen** [comment]
That did work, but I noticed that I no longer am able to customize the text for the loading stuff.  Is that correct?

**Comment by nickspoons** [comment]
You're right! There was a bug there, should be fixed now, with e1a53d3

You can also use `sharpenup#statusline#GetStatus().State` for custom handling in your `LightlineSharpenUp` function:

- `-1`: Ignore (not a .cs buffer)
- `0`: Loading
- `1`: Ready
- `2`: Dead
